assets/chef-recipes/cookbooks/docker/recipes/default.rb

```diff
@@ -54,8 +54,24 @@
     # The current version is always installed
     command 'sudo zypper install -y docker'
   end
-  execute 'Enable and start Docker service' do
+  execute 'Enable Docker service' do
     command 'sudo systemctl enable docker'
+  end
+  execute 'Start Docker service' do
     command 'sudo systemctl start docker'
   end
 end
+
+if node[:platform_family] == 'debian'
+  user = ENV['SUDO_USER']
+  home_dir = Dir.home(user)
+  gnupg_dir = File.join(home_dir, '.gnupg')
+  execute 'Chown ~.gnupg/ directory to sudo user' do
+    command "sudo chown #{user}:#{user} #{gnupg_dir}"
```

Comment:
You should instead of two commands just call the `chown` with the `-R` flag.

Also remove the check for the platform_family. You already have a better check with `only_if` statement.